packages/xo-web/src/common/utils.js

```diff
@@ -371,13 +369,23 @@ export const resolveResourceSets = resourceSets =>
 //   id: 42,
 // }, 32)
 //
-// newString === 'foo_COPY_foo_42_32'
+// newString === 'foo_COPY_foo_42_32%'
 // ```
 export function buildTemplate(pattern, rules) {
-  const regExp = new RegExp(join(map(keys(rules), escapeRegExp), '|'), 'g')
+  const syntaxesToReplace = []
+  Object.keys(rules).forEach(syntax =>
+    syntaxesToReplace.push(escapeRegExp(syntax), escapeRegExp(`\\${syntax}`))
+  )
+  syntaxesToReplace.push(escapeRegExp('\\\\'))
```

Comment:
Wouldn't it be better to add it to the rules instead of relying on the next step's implementation?